README.md

```diff
@@ -46,3 +46,172 @@ note: the UI currently uses code directly from github, pointing to the ts-client
+| stable-x86_64-apple-darwin (1.69.0) | broken        |         1.14.x |
+| 1.68.2-x86_64-apple-darwin          | broken        |         1.14.x |
+| 1.67.1-x86_64-apple-darwin          | broken        |         1.14.x |
+
+Check/set the version:
+```
+rustup override list
+# WRONG:
+# /Users/username/code/mango-v4               1.69-aarch64-apple-darwin
+
+rustup override set 1.69-x86_64
+# CORRECT:
+# /Users/username/code/mango-v4               1.69-x86_64
+```
+
+The Rust version requirements for *Solana* and *Mango V4* can be found in the respective `rust-toolchain.toml` file.
+
+On *MacOS* its also required to use the x86 toolchain.
+
+#### Failed to create BPF VM: syscall #4389198304 was not registered before bind
+This error results from wrong Rust version!
+
+*Solution*: see [Section about Rust version](#wrong-rust-version)
+
+
+#### attempt to compute 0_usize - 1_usize, which would overflow
+Multiple errors of this style are shown:
+```
+666 | const_assert_eq!(size_of::<AnyNode>(), size_of::<LeafNode>());
+| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ attempt to compute `0_usize - 1_usize`, which would overflow
+```
+
+Note: the assertion error text is misleading, the problem is not the overflow but the inequality of the two constants which cannot be optimized to zero by compiler.
+
+The problem appears on non-x86-targets (typically on MacOS).
+
+*Solution*:
+Inspect your rust version and make sure you use the x86 toolchain - see [Section about Rust version](#wrong-rust-version)
+
+
+#### Failed custom program error: 0x65
+(*anchor-related issue*)
+
+*Solution*:
+Make sure that instruction layout (defined using Anchor) used by client/cpi matches the program version.
+
+
+#### Program failed to complete: Access violation in stack frame 5 at address 0x200005ff8 of size 8 by instruction #6340
+No real solution yet! Please open an issue if you know how to fix this.
+
+*Workaround*:
+Experiment with `#[inline(never)]` on methods.
+
+
+#### instruction tries to borrow reference for an account which is already borrowed
+(*anchor-related issue*)
+
+*Solution*:
+Revisit the account references retrieved by explicit calls to `.load`/`.load_mut`. See doc on [`account_loader.rs`](https://github.com/coral-xyz/anchor/blob/fc9fd6d24b9be84abb2f40e47ed3faf7b11864ae/lang/src/accounts/account_loader.rs#L34).
+
+
+#### Running on-chain program crash silently on VM when using rust logging
+
+*Solution*:
+For Solana programs you must use `msg!` for logging and ***not*** `info!`/`debug!`/`error!`/`warn!`/`trace!` from the `log` crate.
+
+
+#### Exceeded max BPF to BPF call depth of 64 at instruction #4605 (stack size exceeded)
+
+*Solution*: Fix the Rust version - see [Section about Rust version](#wrong-rust-version)
+
+
+#### Program failed to complete: Invoked an instruction with data that is too large (12884928151 > 10240)
+
+*Solution*:
+Fix the Solana version (1.15.2 worked).
```

Comment:
let's not mention a different version here - the readme ideally has only a single version referenced once (otherwise we'll have stale old references later)

Reply:
shouldn't we point out when that was fixed

Reply:
Hmm, maybe the readme isn't the right place? Maybe this should be FAQ.md? To me, fixed issues or outdated info doesn't belong in the readme

Reply:
very true. maybe the writing should go into faq/troubleshooting document?